Source/Basetypes/BigFloat.cs

```diff
@@ -157,6 +157,56 @@ public static BigFloat FromBigDec(BigDec v, int significandSize, int exponentSiz
       return new BigFloat(isNeg, sig, exp, sigSize, expSize);
     }
 
+    [Pure]
+    public static BigFloat FromBVString(String s)
+    {
+      /*
+       * String must be either of the format (fp (bv* *) (bv* *) (bv* *))
+       * or of the special value format: (_ val * *)
+       * Where * indicates an integer value
+       * and val is one of the special values: NaN, +oo, -oo, zero, -zero
+       */
+      BIM sig, exp;
+      int sigSize, expSize;
+      bool isNeg;
+
+      String[] args;
+
+      if (s.IndexOf("_") == 1)
+      {
+        s = s.Substring(0, s.Length - 1);
+        args = s.Split(' ');
+        sigSize = int.Parse(args[3]);
+        expSize = int.Parse(args[2]);
+        if (sigSize <= 0 || expSize <= 0)
+          throw new FormatException("Significand and Exponent sizes must be greater than 0");
+        return new BigFloat(args[1], sigSize, expSize);
+      }
+
+      args = s.Split('_');
+
+      int expIndex = args[2].IndexOf(' ', args[2].IndexOf(' ') + 1);
+      int sigIndex = args[3].IndexOf(' ', args[3].IndexOf(' ') + 1);
+
+      isNeg = args[1][3] == '1';
+      exp = BIM.Parse(args[2].Substring(3, expIndex - 3));
+      expSize = int.Parse(args[2].Substring(expIndex + 1, args[2].Length - expIndex - 4));
+      sig = BIM.Parse(args[3].Substring(3, sigIndex - 3));
+      sigSize = int.Parse(args[3].Substring(sigIndex + 1, args[3].Length - sigIndex - 3));
+      if (sigSize <= 0 || expSize <= 0)
+        throw new FormatException("Significand and Exponent sizes must be greater than 0");
+
+      sig = BIM.Abs(sig); //sig must be positive
```

Comment:
Should you maybe check here that it is indeed positive? And throw an exception otherwise?